lib/util/git/index.ts

```diff
@@ -673,19 +673,44 @@ export async function isBranchModified(branchName: string): Promise<boolean> {
     }
     logger.warn({ err }, 'Error checking last author for isBranchModified');
   }
-  const { gitAuthorEmail } = config;
-  if (
-    lastAuthor === gitAuthorEmail ||
-    config.ignoredAuthors.some((ignoredAuthor) => lastAuthor === ignoredAuthor)
-  ) {
-    // author matches - branch has not been modified
-    logger.debug('branch.isModified() = false');
+  const { gitAuthorEmail, ignoredAuthors } = config;
+
+  const includedAuthors = new Set(committedAuthors);
+
+  if (gitAuthorEmail) {
+    includedAuthors.delete(gitAuthorEmail);
+  }
+
+  for (const ignoredAuthor of ignoredAuthors) {
```

Comment:
are we sure ignoredAuthors is never null or undefined? 🤔

Reply:
Looks like @rarkins was thinking similarly with the code suggestion below, however I'm fairly confident it's always at minimum an empty array:

- https://github.com/renovatebot/renovate/blob/09743429d18b7778a95259de2b67fa3d83c45713/lib/util/git/index.ts#L232
- https://github.com/renovatebot/renovate/blob/09743429d18b7778a95259de2b67fa3d83c45713/lib/util/git/index.ts#L323
- https://github.com/renovatebot/renovate/blob/09743429d18b7778a95259de2b67fa3d83c45713/lib/util/git/types.ts#L33